### PR TITLE
fix: FastifySchemaCompiler signature incompatible with AJV

### LINKED ISSUE #2572: Typescript types for `setValidatorCompiler` mismatch
## 🐛 Bug Report

Typescript typings no longer work for setting ajv as a schema validator (even when using the code sample from the README).

## To Reproduce

```typescript
  const ajv = new Ajv({
    // the fastify defaults (if needed)
    removeAdditional: false,
    useDefaults: true, // replace missing properties and items with the values from corresponding default keyword
    coerceTypes: true, // change data type of data to match type keyword
    allErrors: true, // check for all errors
    nullable: true,
  })
  fastify.setValidatorCompiler(({ schema }) => {
    return ajv.compile(schema) 
  })
```

This yields: 

```
src/setupFastify.ts:74:47 - error TS2322: Type 'ValidateFunction' is not assignable to type 'FastifyValidationResult'.
  Types of property 'errors' are incompatible.
    Type 'ErrorObject[]' is not assignable to type 'FastifySchemaValidationError[]'.
      Type 'ErrorObject' is not assignable to type 'FastifySchemaValidationError'.
        Property 'message' is optional in type 'ErrorObject' but required in type 'FastifySchemaValidationError'.

74   fastify.setValidatorCompiler(({ schema }) => ajv.compile(schema))
                                                 ~~~~~~~~~~~~~~~~~~~

  node_modules/fastify/types/schema.d.ts:35:37
    35 export type FastifySchemaCompiler = (routeSchema: FastifyRouteSchemaDef) => FastifyValidationResult
                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    The expected type comes from the return type of this signature.
```

It would be nice to get an up to date sample on how to configure fastify with different ajv settings.

## Expected behavior

Typescript compilation succeeds 

## Your Environment

- *node version*: 12.13
- *fastify version*: 3.4.1
- *typescript version*: 4.0.3
- *os*: Mac
- *any other relevant information*

### PR DESCRIPTION
**Pull request by L2jLiga**
Closes #2572

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test/types/schema.test-d.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { expectAssignable, expectError } from 'tsd'
 import fastify, { FastifyInstance } from '../../fastify'
+import * as Ajv from 'ajv'
```
Comment (fox1t) [inline test/types/schema.test-d.ts:3]: Namespace should not be callable/newable. Use `import Ajv = require('ajv')` to make it work in all scenarios.

**Review by fox1t — CHANGES_REQUESTED** [review]

File: test/types/schema.test-d.ts (hunk 1 above)
Comment (L2jLiga) [reply test/types/schema.test-d.ts:3]: Sure

**Review by mcollina — APPROVED** [review]
lgtm

**Review by fox1t — APPROVED** [review]
LGTM

**Pull request merged**